parse/mod.go

```diff
@@ -2,40 +2,37 @@ package parse
 
 import (
 	"github.com/1Password/dep-report/models"
-	"encoding/json"
-	"fmt"
-	"os/exec"
+	"github.com/pkg/errors"
+	"golang.org/x/mod/modfile"
+	"io/ioutil"
 	"strings"
 )
-//ParseModules runs the `go list` command and formats the output for further processing
-func ParseModules() ([]models.Module, error) {
-	var modArray []models.Module
 
-	goMod, err := exec.Command("go", "list", "-m", "-mod=mod", "-json", "all").Output()
+//ParseModules parses the go.mod file and formats the output for further processing
+func ParseModules(filepath string) ([]models.Module, error) {
+	modBytes, err := ioutil.ReadFile(filepath)
 	if err != nil {
-		return nil, fmt.Errorf("unable to execute go list command: %v", err)
+		return nil, errors.Wrap(err, "unable to read go.mod")
 	}
-	goModString := string(goMod)
-
-	splitGoMod := strings.SplitAfter(goModString, "}\n")
 
-	for _, mod := range splitGoMod {
-		var module models.Module
-		if mod != "" {
-			err := json.Unmarshal([]byte(mod), &module)
-			if err != nil {
-				return nil, fmt.Errorf("unable to unmarshal module into struct: %w", err)
-			}
-			if module.Version != "" {
-				modArray = append(modArray, module)
-			}
-		}
+	formattedMods, err := modfile.Parse("go.mod", modBytes, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to parse go.mod")
 	}
+	var modArray []models.Module
 
+	for _, mod := range formattedMods.Require {
+		var tempMod models.Module
+		tempMod.Path = mod.Mod.Path
+		tempMod.Version = mod.Mod.Version
```

Comment:
Is this and the previous lines the only things we are using from the `mod` package? If so, how complicated is the action this is taking? 

If is something simple, it seems overkill to bring in 3.5k new lines and 25 files for just this single change. If it isn't, let's keep the library.

Reply:
I would prefer to leave the library in. The logic for parsing the go.mod file isn't super straight forward and the go community has done all of the legwork to ensure this is done correctly. We could reverse engineer it, but I fear that we will run into bugs down the line that the go community has already worked through.

Reply:
I agree, let's use the official library.

Reply:
Solid. I agree. Thanks for considering the question